cpp/models/ide_secir/model.h

```diff
@@ -59,37 +59,41 @@ class Model
     bool check_constraints(ScalarType dt) const
     {
         if (!((int)m_transitions.get_num_elements() == (int)InfectionTransition::Count)) {
-            log_error("A variable given for model construction is not valid. Number of elements in transition vector "
-                      "does not match the required number.");
+            log_error(
+                "Initialization failed. Number of elements in transition vector does not match the required number.");
+            return true;
+        }
+
+        // We have some things that just work out if the last time point is zero, e.g. the first time point in m_populations is set to zero ion the constructor.
```

Comment:
```suggestion
        // We have some things that just work out if the last time point is zero, e.g. the first time point in m_populations is set to zero in the constructor.
```
With respect to PR #979 this can already be removed here in my opinion.